static/js/components/EmploymentTab.js

```diff
@@ -65,56 +79,232 @@ class EmploymentTab extends ProfileTab {
     );
   }
 
-  employmentEntriesForm (index) {
-    let keySet = (key) => ['work_history', index, key];
+  componentWillMount () {
+    const { ui, setWorkHistoryEdit } = this.props;
+    if ( _.isUndefined(ui.workHistoryEdit) ) {
+      setWorkHistoryEdit(true);
+    }
+  }
+
+  toggleWorkHistoryEdit = () => {
+    const { ui, setWorkHistoryEdit } = this.props;
+    setWorkHistoryEdit(!ui.workHistoryEdit);
+  }
+
+  closeWorkDialog = () => {
+    const { setWorkDialogVisibility } = this.props;
+    setWorkDialogVisibility(false);
+  }
+
+  addWorkHistoryEntry = () => {
+    const {
+      updateProfile,
+      profile,
+      setWorkDialogIndex,
+      setWorkDialogVisibility,
+    } = this.props;
+    let clone = Object.assign({}, profile);
+    clone['work_history'] = clone['work_history'].concat(
+      _.cloneDeep(this.blankWorkHistoryEntry)
```

Comment:
I feel like having multiple references to the same object should be legal in the redux store, since those objects are not supposed to be modified. I investigated and it looks like `_.cloneDeep` will clone objects with two or more references but those references all get the same clone. Python does this too in `deepcopy`, it looks like it keep track of already cloned objects to prevent recursion errors.

I'm thinking maybe this would be best handled in the `bound` methods, but cloning `blankWorkHistoryEntry` is probably a simpler solution for now